### PR TITLE
Update EarlyStopping doc string and contrib dependency

### PR DESCRIPTION
**Pull request by vfdev-5**
- Same question was asked multiple times #258 #271 
- Fix problem when build the docs for contrib metrics

### CODE REVIEW

**Comment by jasonkriss** [comment]
@vfdev-5 maybe we should add a comment in the codeblock that says something like
```python
# Note: the handler is attached to an *Evaluator*
```
What do you think?

**Comment by vfdev-5** [comment]
@jasonkriss so the whole block is like this ?
```
    .. code-block:: python

        from ignite.engine import Engine, Events
        from ignite.handlers import EarlyStopping

        def score_function(engine):
            val_loss = engine.state.metrics['nll']
            return -val_loss

        handler = EarlyStopping(patience=10, score_function=score_function, trainer=trainer)
        # Note: the handler is attached to an *Evaluator* (runs one epoch)
        evaluator.add_event_handler(Events.COMPLETED, handler)
```

**Comment by jasonkriss** [comment]
@vfdev-5 👍exactly. What do you think?

**Comment by vfdev-5** [comment]
Yes, I like it. Let me commit the change

**Comment by pbamotra** [comment]
Also, I think trainer=trainer is causing the Early Stopping handler to track the training accuracy when I really want the training to stop when the improvements in validation accuracy vanishes beyond patience.

```
trainer = create_supervised_trainer(model, optimizer, criterion, device=device)

evaluator = create_supervised_evaluator(model,
                                            metrics={'accuracy': CategoricalAccuracy(),
                                                     'nll': Loss(criterion)},
                                            device=device)

def score_function(engine):
        return engine.state.metrics['accuracy']

earlystop_handler = EarlyStopping(patience=5, score_function=score_function, trainer=trainer)
evaluator.add_event_handler(Events.COMPLETED, earlystop_handler)
```

**Am I doing anything wrong here?**

Also, I modified the EarlyStopping handler to print messages

```
    def __init__(self, patience, score_function, trainer):
        assert callable(score_function), "Argument score_function should be a function"
        assert patience > 0, "Argument patience should be positive"
        assert isinstance(trainer, Engine), "Argument trainer should be an instance of Engine"
        self.score_function = score_function
        self.patience = patience
        self.trainer = trainer
        self.counter = 0
        self.best_score = None
        self._logger = logging.getLogger(__name__ + "." + self.__class__.__name__)
        self._logger.addHandler(logging.NullHandler())

    def __call__(self, engine):
        score = self.score_function(engine)

        if self.best_score is None:
            self.best_score = score
        elif score < self.best_score:
            self.counter += 1
            tqdm.write("EarlyStopping: %i / %i\n" % (self.counter, self.patience))
            if self.counter >= self.patience:
                tqdm.write("EarlyStopping: Stop training\n")
                self.trainer.terminate()
        else:
            tqdm.write("Resetting counter. New best score: {}\t Prev best: {}\n".format(score, self.best_score))
            self.best_score = score
            self.counter = 0
```

And as I see it, the scores it track are the training data scores.

**Comment by vfdev-5** [comment]
@pbamotra could you please post a complete snippet on gist ?

Following your code:
```python
def score_function(engine):
        return engine.state.metrics['accuracy']


class EarlyStopping:
     # ....
     def __call__(self, engine):
        score = self.score_function(engine)


evaluator.add_event_handler(Events.COMPLETED, earlystop_handler)
```

So as `earlystop_handler` is attached to `evaluator` so when `self.score_function(engine)` is called, the `engine` should be `evaluator`. Could you please check it in your code?  Trainer passed in the `EarlyStopping` is just needed to stop the training and nothing else.
HTH
